### PR TITLE
memory compression/decompression support

### PR DESCRIPTION
**Pull request by XinfengZhang**
add some variant fourcc
add memory hint to disable redundant compress/decompress

### CODE REVIEW

**Comment by xhaihao** [comment]
@XinfengZhang could you rebase your patch upon the latest master branch?

File: va/va.h [hunk 1]
```diff
@@ -1324,9 +1326,28 @@ typedef struct _VASurfaceAttribExternalBuffers {
...
+ * | VA_SURFACE_EXTBUF_DESC_COMPRESSED is true
+ */
+typedef struct _VASurfaceAttribCompressInfo{
+    /** Format modifier applied to this surfaces. */
+    uint64_t drm_format_modifier;
+    /** compressionn data offset */
+    uint32_t aux_plane_index;
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t va_reserved[VA_PADDING_LOW];
+}VASurfaceAttribCompressInfo;
```
[the remark this answers is not in the document]
Comment (XinfengZhang) [reply va/va.h]: @fhvwy  hi , it's for MMC import. the compressed surface from other component is conveyed into media driver. media driver need to know the compression information.

**Comment by XinfengZhang** [comment]
yes, exporting will still be used, even it is encrypted surfaces, drm_format_modifier should be used to indicate it.

File: va/va.h [hunk 2]
```diff
@@ -1324,9 +1326,28 @@ typedef struct _VASurfaceAttribExternalBuffers {
 /** \brief Surface used for export to third-party APIs, e.g. via
  *  vaExportSurfaceHandle(). */
 #define VA_SURFACE_ATTRIB_USAGE_HINT_EXPORT 	0x00000020
+/** \brief Surface is read often from middleware/application
+ * with this hint, no compression was apllied the surface*/
+#define VA_SURFACE_ATTRIB_USAGE_HINT_SW_READ_OFEN  0x00000040
```
Comment (sreerenjb) [inline va/va.h]: "%s/OFEN/OFTEN" 
Is it a typo or intentional? :)

**Comment by fhvwy** [comment]
For the DRM import with format modifier case, I've made an alternative suggestion in <https://github.com/intel/libva/pull/228> which adds support for it using the same structure as export.  I think that the approach there is slightly more consistent than this method which adds a different structure for import only?

I've used DRM_FORMAT_RGBA / I915_FORMAT_MOD_Y_TILED_CCS in the example there - if you have a specific different case in mind here then please do say and I will update the example there to match it.

**Comment by fhvwy** [comment]
> import will use VADRMPRIMESurfaceDescriptor instead of VASurfaceAttribExternalBuffers when the VASurfaceAttribMemoryType == PRIME_2 , right?

Yes, precisely that.  I've updated the comment on `VASurfaceAttribExternalBufferDescriptor` to clarify this point.

**Pull request closed without merging**